matchers/README.md

```diff
@@ -0,0 +1,160 @@
+# OnPar Matchers
+
+OnPar provides a set of minimalistic matchers to get you started.
+However, the intention is to be able to write your own custom matchers so that
+your code is more readable.
+
+
+## Matchers List
+- [String Matchers](#string-matchers)
+- [Logical Matchers](#logical-matchers)
+- [Error Matchers](#error-matchers)
+- [Channel Matchers](#channel-matchers)
+- [Collection Matchers](#collection-matchers)
+- [Other Matchers](#other-matchers)
+
+
+## String Matchers
+### StartWith
+StartWithMatcher accepts a string and succeeds if the actual string starts with
+the expected string.
+
+```go
+Expect(t, "foobar").To(StartWith("foo"))
+```
+
+### EndWith
+EndWithMatcher accepts a string and succeeds if the actual string ends with
+the expected string.
+
+```go
+Expect(t, "foobar").To(EndWith("bar"))
+```
+
+### ContainSubstring
+ContainSubstringMatcher accepts a string and succeeds if the expected string is a
+sub-string of the actual.
+
+```go
+Expect(t, "foobar").To(ContainSubstring("ooba"))
+```
+
+## Logical Matchers
+### Not
+NotMatcher accepts a matcher and will succeed if the specified matcher fails.
+
+```go
+Expect(t, false).To(Not(BeTrue()))
+```
+
+### BeAbove
+BeAboveMatcher accepts a numerical value. It succeeds if the actual is greater
+than the expected.
+
+```go
+Expect(t, 100).To(BeAbove(99))
+```
+
+### BeBelow
+BeBelowMatcher accepts a numerical value. It succeeds if the actual is
+less than the expected.
+
+```go
+Expect(t, 100).To(BeBelow(101))
+```
+
+### BeFalse
+BeFalseMatcher will succeed if actual is false.
+
+```go
+Expect(t, 2 == 3).To(BeFalse())
+```
+
+### BeTrue
+BeTrueMatcher will succeed if actual is true.
+
+```go
+Expect(t, 2 == 2).To(BeTrue())
+```
+
+### Equal
+EqualMatcher performs a DeepEqual between the actual and expected.
+
+```go
+Expect(t, 42).To(Equal(42))
+```
+
+## Error Matchers
+### HaveOccurred
+HaveOccurredMatcher will succeed if the actual value is a non-nil error.
+
+```go
+Expect(t, err).To(HaveOccurred())
+
+Expect(t, nil).To(Not(HaveOccurred()))
+```
+
+## Channel Matchers
+### Always
```

Comment:
I don't think this belongs in the channel matchers category. Maybe asynchronous tests along with ViaPolling?